src/Chapter2.hs

```diff
@@ -349,8 +351,9 @@ Implement a function that returns only the first half of a given list.
 "b"
 -}
 -- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
-
+firstHalf :: [a] -> [a]
+firstHalf [] = []
+firstHalf l = take (length l `div` 2) l
```

Comment:
Great! Note that `take` covers case with the empty list, so no need to have this separately 🤠